docs/book/src/clusterctl/overview.md

```diff
@@ -22,9 +22,13 @@ Below you can find a list of main clusterctl commands:
 
 For the full list of clusterctl commands please refer to [commands](commands/commands.md).
 
+### Avoiding GitHub rate limiting
+
+For normal usage free tier is enough but for extensive usage it might hit the rate limit. To avoid rate limiting for the public repos set the `GITHUB_TOKEN` environment variable. To generate a token [follow this](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) documentation. The token only needs `repo` scope for clusterctl.
```

Comment:
```suggestion
While using providers hosted on GitHub, clusterctl is calling GitHub API which are rate limited; for normal usage free tier is enough but when using clusterctl extensively users might hit the rate limit. 

To avoid rate limiting for the public repos set the `GITHUB_TOKEN` environment variable. To generate a token [follow this](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) documentation. The token only needs `repo` scope for clusterctl.
```